kube-runtime/src/utils/stream_subscribable.rs

```diff
@@ -0,0 +1,58 @@
+use core::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use futures::{stream, Stream, TryStream};
+use pin_project::pin_project;
+use tokio::sync::broadcast;
+
+#[pin_project]
+/// todo - docs
+#[must_use = "streams do nothing unless polled"]
+pub struct StreamSubscribable<S>
```

Comment:
I would call this `StreamSubscribe` to be more in line with normal, unconjugated naming of https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html